onnx/compose.py

```diff
@@ -0,0 +1,256 @@
+            raise ValueError(f"Output {g1_out_name} not present in g1")
+        if g2_in_name not in g2_ins:
+            raise ValueError(f"Input {g2_in_name} not present in g2")
+
+    g = GraphProto()
+
+    g.node.extend(g1.node)
+    g.node.extend(g2.node)
+
+    # Connecting outputs of the first graph with the inputs of the second
+    for g1_out_name, g2_in_name in io_map:
+        for node in g.node:
+            for index, name in enumerate(node.input):
+                if name == g2_in_name:
+                    node.input[index] = g1_out_name
+
+    g.input.extend(g1.input)
+    g.input.extend(
+        [item for item in g2.input if item.name not in io_map_g2_ins])
+
+    g.output.extend(
+        [item for item in g1.output if item.name not in io_map_g1_outs])
+    g.output.extend(g2.output)
+
+    g.initializer.extend(g1.initializer)
+    g.initializer.extend(g2.initializer)
+
+    g.sparse_initializer.extend(g1.sparse_initializer)
+    g.sparse_initializer.extend(g2.sparse_initializer)
+
+    g.value_info.extend(g1.value_info)
+    g.value_info.extend(g2.value_info)
+
+    g.name = name if name is not None else "_".join([g1.name, g2.name])
+
+    if doc_string is None:
+        doc_string = f"Graph combining {g1.name} and {g2.name}\n" + \
+            g1.name + "\n\n" + g1.doc_string + "\n\n" + g2.name + "\n\n" + g2.doc_string
+    g.doc_string = doc_string
+
+    return g
+
+
+def merge_models(
+        m1,  # type: ModelProto
+        m2,  # type: ModelProto
+        io_map,  # type: List[Tuple[Text, Text]]
+        name=None,  # type: Optional[Text]
+        doc_string=None,  # type: Optional[Text]
+        producer_name='onnx.compose',  # type: Optional[Text]
+        ir_version=None,  # type: Optional[int]
+):  # type: (...) -> ModelProto
+    """Combines two ONNX models into a single one.
+
+    The combined model is defined by connecting the specified set of outputs/inputs. Those inputs/outputs
+    not specified in the io_map argument will remain as inputs/outputs of the combined model.
+
+    Arguments:
+        m1 (ModelProto): First model
+        m2 (ModelProto): Second model
+        io_map (list of pairs of string): The pairs of names [(out0, in0), (out1, in1), ...]
+                                          representing outputs of the first graph and inputs of the second
+                                          to be connected
+        name (string): Optional name for the combined graph
+                       By default, the name is g1.name and g2.name concatenated with an undescore delimiter
+        doc_string (string): Optional docstring for the combined graph
+                             If not provided, a default docstring with the concatenation of g1 and g2 docstrings is used
+        producer_name (string): Optional producer name for the combined model. Default: 'onnx.compose'
+        ir_version (int): Optional target IR version. By default the highest IR version from the two graphs will be used
```

Comment:
I am contemplating whether we should make the ir version requirement strict meaning both models should have same IR version. 
we may see issues if one of the models (belonging to older ir version) includes a deleted field.

Reply:
Updated. Now both models should have the same IR version